### ISSUE TITLE
empty ssh-agent cause NPE

### ISSUE THREAD

**Issue by retrogradeorbit** [epiccastle/spire#10]
ssh-agent is active but no keys are added

```
Exception in thread "main" java.lang.NullPointerException
	at clojure.lang.Numbers.bitOpsCast(Numbers.java:1164)
	at clojure.lang.Numbers.and(Numbers.java:1632)
	at spire.pack$unpack_int.invokeStatic(pack.clj:20)
	at spire.pack$unpack_int.invoke(pack.clj:18)
	at spire.pack$decode_string.invokeStatic(pack.clj:31)
	at spire.pack$decode_string.invoke(pack.clj:27)
	at spire.ssh_agent$read_identity.invokeStatic(ssh_agent.clj:20)
	at spire.ssh_agent$read_identity.invoke(ssh_agent.clj:17)
	at spire.ssh_agent$decode_identities.invokeStatic(ssh_agent.clj:32)
	at spire.ssh_agent$decode_identities.invoke(ssh_agent.clj:24)
	at spire.ssh_agent$request_identities.invokeStatic(ssh_agent.clj:58)
	at spire.ssh_agent$request_identities.invoke(ssh_agent.clj:46)
	at spire.ssh_agent$make_identity_repository$fn__9112.invoke(ssh_agent.clj:127)
	at spire.ssh_agent.proxy$java.lang.Object$IdentityRepository$d64b4350.getIdentities(Unknown Source)
	at com.jcraft.jsch.UserAuthPublicKey.start(UserAuthPublicKey.java:39)
	at com.jcraft.jsch.Session.connect(Session.java:470)
	at com.jcraft.jsch.Session.connect(Session.java:183)
	at spire.transport$connect.invokeStatic(transport.clj:24)
	at spire.transport$connect.invoke(transport.clj:12)
	at spire.transport$open_connection$fn__9230.invoke(transport.clj:35)
	at clojure.core$update.invokeStatic(core.clj:6196)
	at clojure.core$update.invoke(core.clj:6188)
	at clojure.lang.Atom.swap(Atom.java:65)
	at clojure.core$swap_BANG_.invokeStatic(core.clj:2354)
	at clojure.core$swap_BANG_.invoke(core.clj:2345)
	at spire.transport$open_connection.invokeStatic(transport.clj:31)
	at spire.transport$open_connection.invoke(transport.clj:29)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:665)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:332)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:321)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:384)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:368)
	at sci.impl.interpreter$eval_let$fn__1082.invoke(interpreter.cljc:53)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:51)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:48)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:342)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:321)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:384)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:368)
	at sci.impl.interpreter$eval_do$fn__1253.invoke(interpreter.cljc:314)
	at sci.impl.interpreter$eval_do.invokeStatic(interpreter.cljc:314)
	at sci.impl.interpreter$eval_do.invoke(interpreter.cljc:310)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:337)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:321)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:384)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:368)
	at sci.impl.interpreter$eval_try.invokeStatic(interpreter.cljc:217)
	at sci.impl.interpreter$eval_try.invoke(interpreter.cljc:213)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:382)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:368)
	at sci.impl.interpreter$eval_let$fn__1087.invoke(interpreter.cljc:59)
	at clojure.core$map$fn__5851.invoke(core.clj:2753)
	at clojure.lang.LazySeq.sval(LazySeq.java:42)
	at clojure.lang.LazySeq.seq(LazySeq.java:51)
	at clojure.lang.LazySeq.next(LazySeq.java:80)
	at clojure.lang.RT.next(RT.java:709)
	at clojure.core$next__5371.invokeStatic(core.clj:64)
	at clojure.core$last__5413.invokeStatic(core.clj:268)
	at clojure.core$last__5413.invoke(core.clj:267)
	at sci.impl.interpreter$eval_let.invokeStatic(interpreter.cljc:59)
	at sci.impl.interpreter$eval_let.doInvoke(interpreter.cljc:48)
	at clojure.lang.RestFn.applyTo(RestFn.java:142)
	at clojure.core$apply.invokeStatic(core.clj:667)
	at clojure.core$apply.invoke(core.clj:660)
	at sci.impl.interpreter$eval_call.invokeStatic(interpreter.cljc:342)
	at sci.impl.interpreter$eval_call.invoke(interpreter.cljc:321)
	at sci.impl.interpreter$interpret.invokeStatic(interpreter.cljc:384)
	at sci.impl.interpreter$interpret.invoke(interpreter.cljc:368)
	at sci.impl.interpreter$eval_form.invokeStatic(interpreter.cljc:422)
	at sci.impl.interpreter$eval_form.invoke(interpreter.cljc:413)
	at sci.impl.interpreter$eval_string_STAR_.invokeStatic(interpreter.cljc:432)
	at sci.impl.interpreter$eval_string_STAR_.invoke(interpreter.cljc:425)
	at sci.impl.interpreter$eval_string.invokeStatic(interpreter.cljc:442)
	at sci.impl.interpreter$eval_string.invoke(interpreter.cljc:438)
	at sci.core$eval_string.invokeStatic(core.cljc:163)
	at sci.core$eval_string.invoke(core.cljc:131)
	at spire.core$evaluate.invokeStatic(core.clj:41)
	at spire.core$evaluate.invoke(core.clj:40)
	at spire.core$_main$fn__14851.invoke(core.clj:64)
	at spire.core$_main.invokeStatic(core.clj:63)
	at spire.core$_main.doInvoke(core.clj:49)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at spire.core.main(Unknown Source)

```

**Comment by retrogradeorbit** [comment]
fixed in 8d7a3679393c74173dcb13eec7db756d534ff3ce

**Comment by retrogradeorbit** [comment]
released in https://github.com/epiccastle/spire/releases/tag/v0.1.0-alpha.1
